aws-py-oidc-provider/README.md

```diff
@@ -0,0 +1,126 @@
+
+This example is an automation of the process detailed in the [AWS documentation for creating an OIDC provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html). This automation will create OIDC configuration between Pulumi Cloud and AWS.
+
+## Prerequisites
+
+* [Install Pulumi](https://www.pulumi.com/docs/get-started/install/)
+* [Configure Pulumi to Use AWS](https://www.pulumi.com/docs/intro/cloud-providers/aws/setup/) (if your AWS CLI is configured, no further changes are required)
+
+## Running the Example
+
+Clone [the examples repo](https://github.com/pulumi/examples/tree/master/aws-py-oidc-provider) and navigate to the folder for this example.
+
+```bash
+git clone https://github.com/pulumi/examples.git
+cd examples/aws-py-oidc-provider-pulumi-cloud
+```
+
+Next, to deploy the application and its infrastructure, follow these steps:
+
+1. Create a new stack, which is an isolated deployment target for this example:
+
+    ```bash
+    $ pulumi stack init dev
+    ```
+
+1. Set your Pulumi organization name and desired AWS region:
+
+    ```bash
+    pulumi config set pulumiOrg <your-pulumi-org-name> # replace with your Pulumi organization name
+    pulumi config set aws:region us-east-1 # any valid AWS region will work
+    ```
+
+1. Install requirements.
+
+    ```bash
+    pip3 install -r requirements.txt
+    ```
+
+1. Run `pulumi up`. 
+
+    ```bash
+    $ pulumi up -y
+    Updating (dev)
+    
+         Type                              Name              Status              Info
+     +   pulumi:pulumi:Stack               oidc-python-dev   created (4s)        8 messages
+     +   ├─ aws:iam:OpenIdConnectProvider  oidcProvider      created (0.78s)     
+     +   └─ aws:iam:Role                   oidcProviderRole  created (0.75s)     
+    
+    Diagnostics:
+      pulumi:pulumi:Stack (oidc-python-dev):
+        Forming configuration document URL...
+        Extracting domain name from jwks_uri...
+        Retrieving OpenSSL certificates (this will take some time)...
+        Retrieving last OpenSSL certificate...
+        Saving certificate to file...
+        Retrieving certificate thumbprint...
+        Creating OIDC provider...
+        Creating Provider IAM role...
+    
+    Outputs:
+        OidcProviderRoleArn: "arn:aws:iam::219544202541:role/oidcProviderRole-c368d93"
+    
+    Resources:
+        + 3 created
+    ```
+## Validating the OIDC Configuration
+
+This next section will walk you through validating your OIDC configuration using [Pulumi ESC](https://www.pulumi.com/docs/pulumi-cloud/esc/). Start by [creating a new Pulumi ESC environment](https://www.pulumi.com/docs/pulumi-cloud/esc/get-started/#create-an-environment). Then, add the following environment definition, replacing the value of `roleArn` with the value of the `OidcProviderRoleArn` from your stack outputs.
+
```

Comment:
Wanna get fancy and `echo` a herestring?

Syntax may not be exactly right, but basically:

```bash
echo > environment.yaml <<<EOF
# ...
          roleArn: $(pulumi stack output OidcProviderRoleArn)
EOF
```

That would make it a little easier for the user. Or heck - just spit out the sample environment as an output.

Reply:
I do like the idea of generating the template as an output. If I can figure out how to get it to print in a usable way before COB tomorrow, i will add that.